model-engine/model_engine_server/infra/gateways/live_sync_model_endpoint_inference_gateway.py

```diff
@@ -186,12 +186,22 @@ async def predict(
         except UpstreamServiceError as exc:
             logger.error(f"Service error on sync task: {exc.content!r}")
             try:
+                # Try to parse traceback from the response, fallback to just return all the content if failed
                 error_json = orjson.loads(exc.content.decode("utf-8"))
-                result_traceback = (
-                    error_json.get("detail", {}).get("traceback")
-                    if isinstance(error_json, dict)
-                    else None
-                )
+                if not isinstance(error_json, dict):
+                    result_traceback = None
+                else:
+                    if "result" in error_json:
+                        error_json = error_json["result"]
+                        if not isinstance(error_json, dict):
+                            error_json = orjson.loads(error_json)
```

Comment:
Can this fail?

Reply:
if this fails it'll go to L209 and we'll return everything unpacked